### ISSUE TITLE
test_network_topology_strategy_counters: AssertionError: Failed to read value from sufficient number of nodes, required 1 but got 0

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylla#9631]
Seen in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-release/996/testReport/consistency_test/TestAccuracy/test_network_topology_strategy_counters/
Scylla version 9d4262e264efd6d92de9c40545f59c85c840de0d
and reproduced locally.
```
Error Message
Look for an exception above
Stacktrace
Traceback (most recent call last):
  File "/usr/lib64/python3.7/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/usr/lib64/python3.7/unittest/case.py", line 645, in run
    testMethod()
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/consistency_test.py", line 760, in test_network_topology_strategy_counters
    TestAccuracy.Validation.validate_counters, self.nodes, self.rf.values(), combinations),
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/consistency_test.py", line 572, in _run_test_function_in_parallel
    assert False, "Look for an exception above"
AssertionError: Look for an exception above
Standard Error
Traceback (most recent call last):
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/consistency_test.py", line 545, in run
    valid_fcn(v)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/consistency_test.py", line 529, in validate_counters
    check_all_sessions(s, n, c)
  File "/jenkins/workspace/scylla-master/dtest-release/scylla-dtest/consistency_test.py", line 522, in check_all_sessions
    % (write_nodes, num, n, val)
AssertionError: Failed to read value from sufficient number of nodes, required 1 but got 0 - [482, 4]
```

There are no errors in the node logs.

I'm bisecting to see when the regression started.

**Comment by bhalevy** [comment]
Oddly, git bisect points at 52feb4146839522975451a44f94c8e1cddc8bb38
Cc @raphaelsc

**Comment by bhalevy** [comment]
@raphaelsc the following failure could be related:
https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-debug-random/630/testReport/junit/alternator_tests/AlternatorTest/test_slow_query_logging/
```
Critical errors found: [('node4', ["scylla: compaction/compaction_manager.cc:1000: future<> compaction_manager::remove(column_family *): Assertion `std::find_if(_tasks.begin(), _tasks.end(), [cf] (auto& task) { return task->compacting_cf == cf; }) == _tasks.end()' failed.", "scylla: compaction/compaction_manager.cc:1000: future<> compaction_manager::remove(column_family *): Assertion `std::find_if(_tasks.begin(), _tasks.end(), [cf] (auto& task) { return task->compacting_cf == cf; }) == _tasks.end()' failed.", 'Aborting on shard 1.'])]
```

**Comment by raphaelsc** [comment]
looking into this...

**Comment by bhalevy** [comment]
FWIW, the test passes for me in dev mode with https://github.com/scylladb/scylla/pull/9636
Retrying to make sure it isn't a fluke....

**Comment by slivne** [comment]
57e967968518d69635ef53ed8e7005354ab03a67 was merged - closing the issue 

@raphaelsc / @bhalevy do we need to backport 57e967968518d69635ef53ed8e7005354ab03a67 ?

**Comment by bhalevy** [comment]
> [57e9679](https://github.com/scylladb/scylla/commit/57e967968518d69635ef53ed8e7005354ab03a67) was merged - closing the issue

thanks

> @raphaelsc / @bhalevy do we need to backport [57e9679](https://github.com/scylladb/scylla/commit/57e967968518d69635ef53ed8e7005354ab03a67) ?

I think it's too risky to backport so we should consider it only if it proves to fix a blocking issue in 4.6

**Comment by bhalevy** [comment]
and the actual commit in master was 1e2ecd282aa5467538b1e9814a2183038cf7776b eventually
